cmd/qgb/common/helpers.go

```diff
@@ -0,0 +1,110 @@
+package common
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	common2 "github.com/celestiaorg/orchestrator-relayer/cmd/qgb/keys/p2p"
+	"github.com/celestiaorg/orchestrator-relayer/helpers"
+	"github.com/celestiaorg/orchestrator-relayer/p2p"
+	"github.com/celestiaorg/orchestrator-relayer/rpc"
+	keystore2 "github.com/ipfs/boxo/keystore"
+	ds "github.com/ipfs/go-datastore"
+	"github.com/libp2p/go-libp2p/core/peer"
+	tmlog "github.com/tendermint/tendermint/libs/log"
+)
+
+// NewTmAndAppQuerier helper function that creates a new TmQuerier and AppQuerier and registers their stop functions in the
+// stopFuncs slice.
+func NewTmAndAppQuerier(logger tmlog.Logger, tendermintRPC string, celesGRPC string, stopFuncs []func() error) (*rpc.TmQuerier, *rpc.AppQuerier, error) {
+	// load app encoding configuration
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+
+	// creating tendermint querier
+	tmQuerier := rpc.NewTmQuerier(tendermintRPC, logger)
+	err := tmQuerier.Start()
+	if err != nil {
+		return nil, nil, err
+	}
+	stopFuncs = append(stopFuncs, func() error {
+		err := tmQuerier.Stop()
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+
+	// creating the application querier
+	appQuerier := rpc.NewAppQuerier(logger, celesGRPC, encCfg)
+	err = appQuerier.Start()
+	if err != nil {
+		return nil, nil, err
+	}
+	stopFuncs = append(stopFuncs, func() error {
+		err = appQuerier.Stop()
+		if err != nil {
+			return err
+		}
+		return nil
+	})
```

Comment:
are trying to return this func to the user?

if so, then we have to accept a pointer to a slice or return one afaiu

Reply:
we are returning it no? or you mean returning the pointer?

Reply:
ahh I think when I reviewed there were unpushed commits maybe? for whatever reason, I was reviewing 4bded402c0a0dcf3ecd299b08b8575eb7c5493b8, but you are correct, it was fixed!